lightning-invoice/src/de.rs

```diff
@@ -1175,4 +1175,53 @@ mod test {
 			)
 		)
 	}
+
+	#[test]
+	fn test_deser_long() {
+		use crate::Bolt11Invoice;
+
+		#[track_caller]
+		fn parse_ok(invoice_str: &str) {
+			let invoice = Bolt11Invoice::from_str(invoice_str).unwrap();
+			let invoice_str2 = invoice.to_string();
+			if invoice_str != invoice_str2 {
+				panic!(
+					"Invoice does not roundtrip: invoice_str != invoice_str2\n\
+					  invoice_str: {invoice_str}\n\
+					 invoice_str2: {invoice_str2}\n\
+					 \n\
+					 {invoice:?}"
+				);
+			}
+		}
+
+		parse_ok("lnbc10000000000000000010p1qqqqqqqdtuxpqkzq8sjzqgps4pvyczqq8sjzqgpuysszq0pyyqsrp2zs0sjzqgps4pxrcfpqyqc2slpyyqsqsv9gwz59s5zqpqyps5rc9qsrs2pqxz5ysyzcfqgysyzs0sjzqgqq8sjzqgps4pxqqzps4pqpssqgzpxps5ruysszqrps4pg8p2zgpsc2snpuysszqzqsgqvys0pyyqsrcfpqyqvycv9gfqqrcfpqyq7zggpq8q5zqyruysszqwpgyqxpsjqsgq7zggpqps7zggpq8sjzqgqgqq7zggpqpq7zggpq8q5zqqpuysszq0pyyqsqs0pyyqspsnqgzpqpqlpyyqsqszpuysszqyzvzpvysrqq8sjzqgqvrp7zggpqpqxpsspp5mf45hs3cgphh0074r5qmr74y82r26ac4pzdg4nd9mdmsvz6ffqpssp5vr4yra4pcv74h9hk3d0233nqu4gktpuykjamrafrdpuedqugzh3q9q2sqqqqqysgqcqrpqqxq8pqqqqqqnp4qgvcxpme2q5lng36j9gruwlrtk2f86s3c5xmk87yhvyuwdeh025q5r9yqwnqegv9hj9nzkhyxaeyq92wcrnqp36pyrc2qzrvswj5g96ey2dn6qqqqqqqqqqqqqqqqqqqqqqqqqqqqqp9a5vs0t4z56p64xyma8s84yvdx7uhqj0gvrr424fea2wpztq2fwqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqmy9qqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqqpcnsxc32du9n7amlypuhclzqrt6lkegq0v3r7nczjv9tv30z7phq80r3dm7pvgykl7gwuenmem93h5xwdwac6ngsmzqc34khrg3qjgsq6qk6lc");
```

Comment:
Could you comment for each of these invoices what they're uniquely testing? Also curious how these were generated?

Reply:
Yeah these are failing outputs from this <https://github.com/lexe-app/lexe-public/blob/4bc7018307e5221e1e1ee8b17ce366338fb11a16/common/src/ln/invoice.rs#L183> proptest generator. Ultimately the issue was just the new max length error, but I also want to ensure this upstream testsuite is successfully roundtripping these weird invoices.